src/khepri_condition.erl

```diff
@@ -290,6 +290,9 @@
 %% This is the same as {@link keep_while()} but the paths in the map keys were
 %% converted to native paths if necessary.
 
+-type re_mp() :: tuple().
+%% `mp()` type defined by {@link re} but not exported.
```

Comment:
The type is called `mp()` in `re`. It is documented but not exported unfortunately. I used `re_mp()` as the closest equivalent to `re:mp()` if it existed. The content of the type is basically opaque to Khepri otherwise.

Do you think it would still make more sense to use our own name?

Reply:
Hmm yeah, I think `re`'s name for the type is sort of cryptic too 😄

I would still prefer something that isn't abbreviated

Reply:
```suggestion
-type re_mp() :: tuple().
%% `mp()' type defined by {@link re} but not exported.
```

Reply:
I changed the type and fixed the spec (which was semantically incorrect in fact). So having a proper name helps to find bugs in the end, thank you for pushing in that direction!

Reply:
Looks great, I like it!